### DISCUSSION TITLE
'Relationship' table (or similar) for miscellaneous relationships should be added

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5643] [Feature Requests]
There needs to be a way to add miscelaneous, generic relationships to CO records or others would be very helpful.

This should include a name, remarks, and weblink field to allow attaching URLs or links to miscellaneous sites that contain data about the specimen or other records that are related to the specimen in some way.

This could be used in situations like this:

**Destination** - URL - Description

**Song.com** - http://song.com/song/1231 - Bird song
**Photos.com** - http://photos.com/photo/1231 - Photo of the bird externally
**Place.com** - http://place.com/place/1231 - Location we found it in

**Requested By:** @acbentley

**Comment by grantfitzsimmons** [comment]
Maybe these should be identified by GUIDs using a broker instead

**Comment by maxpatiiuk** [comment]
This might be a bit hacky, but if you put a URL in the relationship, the front-end can check if the URL refers to an object from the current database. If so, it would format it and allow you to open it.
Else, clicking on the relationship would open the link in a dialog window or in new tab

The problem is that it would be hard to query a relationship that is defined using a URL

**Comment by grantfitzsimmons** [comment]
Again mentioned as something desired by UMich so they can link from CO to any record in the database

**Comment by g1rly-c0d3r** [comment]
linked in https://github.com/specify/specify7/issues/8212
